src/components/PDFView/index.js

```diff
@@ -123,15 +135,31 @@ class PDFView extends Component {
      * @param {Number} height of the page
      * @returns {Number} devicePixelRatio for this page on this platform
      */
-    getDevicePixelRatio(width, height) {
-        const nbPixels = width * height;
-        const ratioHeight = this.props.maxCanvasHeight / height;
-        const ratioWidth = this.props.maxCanvasWidth / width;
-        const ratioArea = Math.sqrt(this.props.maxCanvasArea / nbPixels);
-        const ratio = Math.min(ratioHeight, ratioArea, ratioWidth);
+    const getDevicePixelRatio = useCallback(
```

Comment:
Could we move this function outside? So we don't have to memoize it? And just pass it the props it needs to do the calculations as arguments?